docs/design/custom-resource-metrics-crd.md

```diff
@@ -0,0 +1,211 @@
+# Kube-State-Metrics - CustomResourceMonitor CRD Proposal
+
+
+---
+
+Authors: Catherine Fang (CatherineF-dev@), Christian Schlotter (chrischdi@)
+
+Date: 26. Jun 2023
+
+Target release: v
+
+---
+
+## Table of Contents
+- [Glossary](#glossary)
+- [Problem Statement](#problem-statement)
+- [Goal](#goal)
+- [Status Quo](#status-quo)
+- [Proposal](#proposal)
+   - [New flags](#new-flags)
+   - [CustomResourceMonitor Definition](#customresourcemonitor-definition)
+   - [Watch and Reconcile on CustomResourceMonitor CRs](#watch-and-reconcile-on-customresourcemonitor-crs)
+- [CUJ](#cuj)
+
+
+
+## Glossary
+
+- kube-state-metrics: “Simple service that listens to the Kubernetes API server
+  and generates metrics about the state of the objects”
+- Custom Resource: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources
+- [CustomResourceState](https://github.com/kubernetes/kube-state-metrics/blob/main/docs/customresourcestate-metrics.md) monitoring feature: existing feature which collects custom resource metrics
+
+
+## Problem Statement
+
+1. Using CustomResourceState monitoring feature is not user-friendly. Current ways on configuring CustomResourceState monitoring are:
+* `--custom-resource-state-config "inline yaml (see example)"` or
+* `--custom-resource-state-config-file /path/to/config.yaml`. Either mounted or configmap.
+
+2. Current CustomResourceState monitoring feature doesn't support multiple configuration files.
+
+For example, for a company with 10 teams, each team wants to collect Custom Resource metrics for their owned Custom Resources. 
+
+
+## Goal
+
+A better UX to collect custom resource metrics
+
+## Proposal
+
+Add a custom resource definition (CustomResourceMonitor) which contains customresourcestate.Metrics.
+
+kube-state-metrics watched on CustomResourceMonitor CRs and concatenate these CRs into one config `customresourcestate.Metrics` which has the same content using `--custom-resource-state-config`. 
+
+### New flags
+Apart from existing two flags (`--custom-resource-state-config ` and `--custom-resource-state-config-file`), these three flags will be added: 
+* `--custom_resource_monitor_enabled`: whether watch CustomResourceMonitor CRs or not.
+* `--custom_resource_monitor_labels`: only watch CustomResourceMonitor with label=x. It's used to avoid double custom metrics collection when multiple kube-state-metrics are installed.
+* `--custom_resource_monitor_namespaces`: only watch CustomResourceMonitor under namespaces=x.
```

Comment:
Should we consider a denylist for namespaces as well? 
In case someone has a "platform KSM" (only selected namespaces) and a "product KSM" (everything else)?